### PR TITLE
[TOO-145]GQL Error Adaptor

### PR DESCRIPTION
**Pull request by jottakka**
# 🎫  [TOO-145](https://linear.app/arcadedev/issue/TOO-145/adding-gql-error-interceptor-in-tdk)
- add GraphQL-specific error adapter, expose it, and slot it ahead of HTTP fallback  
- map gql query/transport errors (incl. rate-limit handoff) without logging sensitive data  
- cover adapter + tool chain behavior with new unit tests (pathload, dedupe, edge cases)

closes TOO-145

### CODE REVIEW

File: libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+        # Handle GraphQL Query Errors (HTTP 200 OK but contains 'errors' list)
+        if isinstance(exc, TransportQueryError):
+            return self._handle_query_error(exc)
+
+        # Handle Transport Server Errors (HTTP 500s, 400s from the transport layer)
+        if isinstance(exc, TransportServerError):
+            return self._handle_server_error(exc)
+
+        return None
```
Comment (EricGustin) [inline libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py:97]: Consider adding a catch all for the base `gql.transport.exceptions.TransportError` before returning `None`

File: libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+        developer_message = "GraphQL error"
+        if error_codes:
+            developer_message = f"GraphQL error codes: {', '.join(sorted(set(error_codes)))}"
+
+        return UpstreamError(
+            message=f"Upstream GraphQL error: {joined_message}",
+            status_code=status_code,
+            developer_message=developer_message,
+            extra={
```
Comment (EricGustin) [inline libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py:128]: Please add the request URL as a minimum to the `extra` if possible.

**Review by EricGustin — COMMENTED** [review]
Mostly looks good to me. Package also needs a version bump

File: libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py (hunk 2 above)
Comment (jottakka) [reply libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py:128]: In graphql it always hit the same endpoint for all APIs f `{base_url}/graphql `, so it isn't so directly availabe in the error, so it needed some workaround was needed as it can use different transport libs (aiohttp, httpx, and requests)

File: libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py (hunk 1 above)
Comment (jottakka) [reply libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py:97]: Ok!

File: libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py (hunk 2 above)
Comment (EricGustin) [reply libs/arcade-tdk/arcade_tdk/providers/graphql/error_adapter.py:128]: understood

**Review by EricGustin — APPROVED** [review]
